File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.py
def solution(price, money, count):
    sum = 0
    for i in range(price, price*count+1, price):
        sum+=i
    return sum - money if money < sum else 0

# return max(0,price*(count+1)*count//2-money)
# 개섹시하다. 
# 등차 수열의 합 공식 활용..
# max() 로 음수나오면 (돈 안부족하면) 0 뽑히게..

File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.java
class Solution {
    public long solution(int price, int money, int count) {
        // 총합 = (1*price + 2*price + ... n*price) = n! * price
        // n! * price - money
        long result = 0;
        for (int i = 1; i <= count; i++) result += i;
        // 단, 금액이 부족하지 않으면 0을 return 하세요.
        long answer = result*price - money;
        return answer < 0 ? 0 : answer;
    }
}